hack/test-extended.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+source ${OS_ROOT}/hack/util.sh
+source ${OS_ROOT}/hack/common.sh
+
+# TODO: Randomize these ports
+export OS_EXTENDED_MASTER=127.0.0.1:54199
+export OS_EXTENDED_ASSETS=127.0.0.1:54299
+
+export BASETMPDIR="/tmp/openshift-extended"
+rm -rf ${BASETMPDIR} && mkdir -p ${BASETMPDIR}
+
+# Setup directories and certificates for 'curl'
+export CERT_DIR="${BASETMPDIR}/cert"
+export CURL_CA_BUNDLE="${CERT_DIR}/master/root.crt"
+export CURL_CERT="${CERT_DIR}/admin/cert.crt"
+export CURL_KEY="${CERT_DIR}/admin/key.key"
+export OPENSHIFT_ON_PANIC=crash
+
+cleanup() {
+    set +e
+    kill $(cat ${BASETMPDIR}/server.pid)
+    rm -rf ${ETCD_DIR} 1>&2 2>/dev/null
+    echo "[INFO][$(date)] Cleanup complete"
+}
+
+start_server() {
+  sudo env "PATH=${PATH}" openshift start \
+    --listen="https://${OS_EXTENDED_MASTER}" \
+    --public-master="https://${OS_EXTENDED_MASTER}" \
+    --master="https://${OS_EXTENDED_MASTER}" \
+    --etcd="http://127.0.0.1:4001" \
+    --nodes=127.0.0.1 \
+    --hostname="127.0.0.1" \
+    --cert-dir="${CERT_DIR}" \
+    --loglevel=3 &> ${BASETMPDIR}/server.log &
+  echo -n $! > ${BASETMPDIR}/server.pid
+}
+
+# Go to the top of the tree.
+cd "${OS_ROOT}"
+
+# Start the Etcd server and indicate it to the test-integration, so the etcd
+# will not be started twice
+start_etcd
+export ETCD_STARTED="1"
+
+# Start OpenShift sever that will be common for 
+echo "[INFO][$(date)] Starting OpenShift test server [${OS_EXTENDED_MASTER}]..."
+start_server
```

Comment:
It'd also be useful to instantiate the docker registry after starting the server.